COBOL Programming Course #4 - Testing/COBOL Programming Course #4 - Testing.md

```diff
@@ -188,178 +186,664 @@ Even more damaging could be the impact of being unable to deliver function rapid
+COBOL Check uses a default directory structure to retrieve the COBOL programs, the test suites, copybooks etc. you can take a look at the wiki to get a clear understanding of the directory structures: https://github.com/openmainframeproject/cobol-check/wiki/Default-Directory-Structure
 
- 
+### The test pyramid
+![](Images/image235.png)
 
-When we speak of testing in this chapter from this point on, we are referring to what would more strictly be called **checking**, rather than testing.
+Cobol-check supports fine-grained unit-testing. Let's provide some contextual information to make the meaning of that clearer. This is a popular level of test automation; lets call it a pyramid or a triangle. The key idea is that we need to test our software at multiple levels of abstraction as we move higher in the diagram, we are testing larger pieces of software lower in the diagram, we are testing smaller pieces of software.
 
- 
+The figure is wide at the bottom and narrow at the top to suggest that we want a large number of small test cases and a smaller number of big test cases. It is preferred because considering how difficult it is to test an entire application with all its external dependencies, environment configuration and test data and to test every combination of inputs and operating conditions at this level would be very tedious and time-consuming. Organizations that do all their testing by running the full system often lack sufficient time to test thoroughly before each release. If we push most of the test cases down to a level, we find we can write more test cases with less effort, so we can exercise each part of the code thoroughly. The test case runs in less time, and each test failure is easier to diagnose and fix.
 
-The key point is that anything that lends itself to automation should be automated, to allow time for these other activities where automation is genuinely not appropriate.  The fact that something is quite difficult to automate should not be used as an excuse for failing to automate it.
 
- 
-### Managing the pipeline when test cases fail
 
-As noted above, if the tests are automated, but checking or reacting to their results is not, then that is only half of the story.  When building a CI/CD pipeline, thought should be given to what happens when test cases fail. 
+### Test suites and test cases
 
- 
+TESTSUITE - Provides a description for a series of test cases. The description is echoed in the output from the test run.
+TESTCASE - identifies a test case. The description is echoed in the output of the test run.
```

Comment:
Might need an extra line separating this two, or we can change this to bullet point